scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -62,35 +67,49 @@ abstract class ScanResultsStorage {
          * [FileBasedStorage] using a [XZCompressedLocalFileStorage] as backend is configured.
          */
         fun configure(config: ScannerConfiguration) {
-            val configuredStorages = listOfNotNull(
-                config.fileBasedStorage,
-                config.postgresStorage
-            )
+            storage = if (config.storages.isNullOrEmpty()) createDefaultStorage()
+            else createCompositeStorage(config)
+            log.info("ScanResultStorage has been configured to ${storage.name}.")
+        }
 
-            require(configuredStorages.size <= 1) {
-                "Only one scan results storage may be configured."
-            }
+        /**
+         * Create a [CompositeStorage] that manages all storages defined in the given [config].
+         */
```

Comment:
naming: How about "compound" instead of "composite"?

Reply:
The name was chosen to refer to the Composite GoF pattern, but I am fine with both names. Which one is the preferred one?

Reply:
As a non-native speaker, I doubt that I am capable of getting the nuances correctly; but after googeling both terms, I have the impression that "compound" is also used in a sense of "complex" and "complicated", which would imply something negative. Therefore, I would rather stick to "composite".